tsdoc/src/beta/DeclarationReference.ts

```diff
@@ -53,31 +53,71 @@ export class DeclarationReference {
     return reference;
   }
 
-  public static makeSafeComponent(text: string): string {
-    const parser: Parser = new Parser(text);
+  /**
+   * Escapes a string for use as a symbol navigation component. If the string contains `!.#~:,"{}()` or starts with
+   * `[`, it is enclosed in quotes.
+   */
+  public static escapeComponentString(text: string): string {
+    if (text.length === 0) {
+      return '""';
+    }
+    const ch: string = text.charAt(0);
+    if (ch === '"' || ch === '[' || !this.isWellFormedComponentString(text)) {
+      return JSON.stringify(text);
+    }
+    return text;
+  }
+
+  /**
+   * Unescapes a string used as a symbol navigation component.
+   */
+  public static unescapeComponentString(text: string): string {
+    if (text.length > 2 && text.charAt(0) === '"' && text.charAt(text.length - 1) === '"') {
+      try {
+        return JSON.parse(text);
+      } catch {
+        throw new SyntaxError(`Invalid Component '${text}'`);
+      }
+    }
+    if (!this.isWellFormedComponentString(text)) {
+      throw new SyntaxError(`Invalid Component '${text}'`);
+    }
+    return text;
+  }
+
+  /**
+   * Determines whether the provided string is a well-formed symbol navigation component string.
+   */
+  public static isWellFormedComponentString(text: string): boolean {
     try {
-        parser.parseComponent();
-        if (parser.eof) {
-            return text;
-        }
+      const parser: Parser = new Parser(text);
+      parser.parseComponent();
+      return parser.eof;
     } catch {
```

Comment:
Ideally we should eliminate this "unexceptional exception".  Throwing exceptions during normal (successful) execution interferes with the "Break on exception" debugging experience, and exception callstack unwinding can impact performance.

It's not important for this PR, though.